### PR TITLE
CEDS-1254: fixed broken CSS and componentised the autocomplete elemen…

### PR DESCRIPTION
**Pull request by mattclark-zerogravit**
…t so it can be re-used elsewhere

### CODE REVIEW

File: app/views/declaration/destination_countries_supplementary.scala.html [hunk 1]
```diff
@@ -37,8 +38,16 @@
 
         @components.page_title(Some("declaration.destinationCountries.title"))
 
+        <style>
```
Comment (PatrykRudnicki) [inline app/views/declaration/destination_countries_supplementary.scala.html]: Just propose to move styles to the different file. We don't want to add those styles to every page with autocomplete.

File: conf/location-autocomplete-canonical-list.json [hunk 2]
```diff
@@ -263,10 +263,6 @@
     "Czechia",
     "country:CZ"
   ],
-  [
-    "Czechoslovakia",
-    "country:CS"
-  ],
   [
     "Denmark",
     "country:DK"
```
Comment (PatrykRudnicki) [inline conf/location-autocomplete-canonical-list.json]: In this file you can find McDonald's Islands or something like this. As I remember in ticket we will have a new list with countries.

File: app/services/model/AutoCompleteDataItem.scala [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package services.model
+
+case class AutoCompleteDataItem(label: String, value: String)
```
Comment (benjaminparker) [inline app/services/model/AutoCompleteDataItem.scala]: Could shorten to AutoCompleteItem

File: app/views/components/autocomplete_field.scala.html [hunk 4]
```diff
@@ -80,7 +74,7 @@
 
         <select id="@(fieldWithoutBrackets)" name="@(field)">
             <option value="">@emptySelectValue</option>
-            @for(option <- countries.map(c => (c.countryCode, c.countryName))) {
+            @for(option <- dataItems.map(d => (d.value, d.label))) {
```
Comment (benjaminparker) [inline app/views/components/autocomplete_field.scala.html]: similarly lose the word data

File: app/views/declaration/destination_countries_supplementary.scala.html (hunk 1 above)
Comment (mattclark-zerogravit) [reply app/views/declaration/destination_countries_supplementary.scala.html]: done

File: conf/location-autocomplete-canonical-list.json (hunk 2 above)
Comment (mattclark-zerogravit) [reply conf/location-autocomplete-canonical-list.json]: covered in another ticket. Needs analysis. I have removed 4 obvious errors. Mcdonalds islands is a legit place.

File: app/services/model/AutoCompleteDataItem.scala (hunk 3 above)
Comment (mattclark-zerogravit) [reply app/services/model/AutoCompleteDataItem.scala]: Changed

File: app/views/components/autocomplete_field.scala.html (hunk 4 above)
Comment (mattclark-zerogravit) [reply app/views/components/autocomplete_field.scala.html]: Changed

**Pull request merged**